### PR TITLE
Snapshot fixes

### PR DESCRIPTION
**Pull request by wwestgarth**
Today's snapshot fixes

Related to #6123

### CODE REVIEW

File: core/execution/market_snapshot.go [hunk 1]
```diff
@@ -161,6 +161,10 @@ func NewMarketFromSnapshot(
 	market.tradableInstrument.Instrument.Product.NotifyOnSettlementPrice(market.settlementPrice)
 	liqEngine.SetGetStaticPricesFunc(market.getBestStaticPricesDecimal)
 
+	if mkt.State == types.MarketStateTradingTerminated {
+		market.tradableInstrument.Instrument.Product.UnsubscribeTradingTerminated(ctx)
+	}
```
Comment (jeremyletang) [inline core/execution/market_snapshot.go:166]: Why not d othe check at line 160, and just not register in the first time?

Comment (wwestgarth) [reply core/execution/market_snapshot.go:166]: 160 is the callback from the instrument into the market. The subscription to the oracles happens deep in the heart of `markets.NewTradableInstrument(ctx, log, mkt.TradableInstrument, oracleEngine)`: https://github.com/vegaprotocol/vega/blob/develop/core/products/future.go#L260

Comment (jeremyletang) [reply core/execution/market_snapshot.go:166]: right makes sense. just feels a bit weird that we register it just to unregister soon after.

Comment (wwestgarth) [reply core/execution/market_snapshot.go:166]: I hear you. I'll do a whole `NewTradableInstrumentFromSnapshot()` type thing.

Comment (jeremyletang) [reply core/execution/market_snapshot.go:166]: ah sounds good.

**Review by ValentinTrinque — APPROVED** [review]

**Pull request merged**